lib/python/pyflyby/_importstmt.py

```diff
@@ -486,11 +487,31 @@ def pretty_print(self, params=FormatParams(),
             tokens.append(t)
         res = s0 + pyfill(s, tokens, params=params)
         if params.use_black:
-            import black
-            mode = black.FileMode()
-            return black.format_str(res, mode=mode)
+            return self.run_black(res, params)
```

Comment:
Can't something along the following work w/o a subprocess ?

```
from black import read_pyproject_toml, find_pyproject_toml
from click import Context, Command

c = Context(Command("black"))
print(read_pyproject_toml(c, None, None))
c.default_map

```

I'm pretty sure you can find some inspiration in https://github.com/akaihola/darker/blob/e96018f086383a2dcfaab6825945fbee08daca2a/src/darker/black_diff.py#L97 that use black by importing it.